src/Runner.Common/JobServer.cs

```diff
@@ -56,6 +60,32 @@ public async Task ConnectAsync(VssConnection jobConnection)
             _hasConnection = true;
         }
 
+        // Refresh connection is best effort. it should never throw exception
+        private async Task RefreshConnectionAsync(Uri jobServerUrl, VssCredentials jobServerCredential, DelegatingHandler[] delegatingHandler)
+        {
+            Trace.Info($"Refresh JobServer VssConnection to get on a different AFD node.");
+
+            VssConnection newConnection = null;
+
+            try
+            {
+                _hasConnection = false;
```

Comment:
i think we don't need to change `_hasConnection` in the `RefreshConnectionAsync` since it will only get set to `true` in line 60 after we get a connection up and running.